Update LiveDNS endpoint and the docs URL

Themimitoof opened the pull request:
Since April 2020 we (Gandi) have deprecated the old LiveDNS endpoint in favor of the new one that is part of our public API project.

This PR update the endpoint but also the link to the new documentation.

cc @grigouze @romuald

Commit:
Update LiveDNS endpoint and the docs URL

Since April 2020 we (Gandi) have deprecated the old LiveDNS endpoint
in favor of the new one that is part of our public API project.

This commit update the endpoint but also the link to the new
documentation.

romuald commented:
Nicely spotted, however this is not enough ^^

The authentication mechanism did change between the 2 endpoints:
you need to authenticate on api.gandi.net with an `Authorization: apikey XXXX` header instead of `X-API-Key: XXXX`

Commit:
Update the authentication mechanism

The new public API now uses `Authentication: Apikey <Apikey>` header to
authenticate requests.

Themimitoof commented:
Thanks @romuald for the review, I just pushed a commit to update this part 😉.

obynio commented:
@Themimitoof the new API is still shown as being in beta and this plugin is being used for production systems. Do you consider your API mature enough to be used in production ?

obynio commented:
@Themimitoof Thank you for your PR, I'll take a look to ensure the endpoints are the same before making the switch. However your new API is still shown as being in beta and this plugin is being used for production systems. Do you believe the new API is mature enough to be used in production ?

obynio commented:
@Themimitoof Thank you for your PR, I'll take a look to ensure the endpoints are the same before making the switch. However your new API is still shown as being in beta and this plugin is being used for production systems. Do you believe the new API is mature enough to be used in production ?

The documentation [explicitly states](https://api.gandi.net/docs/reference/#Introduction) `Please note that this API is currently in BETA testing, so care should be taken when used in production`

obynio reviewed (changes requested):
At a first glance, it appears the `https://api.gandi.net/v5/livedns/domains` endpoint has changed and the attribute `zone_records_href` from the previous API has vanished. 

Unfortunately for some silly reason of mine I was using this attribute here and there to make my requests. Could you replace the `domain.ZoneRecordsHref` currently used to make requests by `domain.DomainRecordsHref` ?

obynio commented:
I see that the API now accepts both rrset names and fully qualified domains names for [this endpoint](https://api.gandi.net/docs/livedns/#v5-livedns-domains-fqdn-records). That's good, it means I we may finally ditch [that hack](https://github.com/libdns/gandi/blob/master/client.go#L20) I'm not so proud of and use either `hello.world.com.` and `hello` to manipulate zones.

obynio reviewed (changes requested):
I see that the when using the `domain_records_href` instead of the `zone_records_href` attribute we can submit both relative and absolute names for [this endpoint](https://api.gandi.net/docs/livedns/#v5-livedns-domains-fqdn-records).

That's good, it means I we may finally ditch [that hack](https://github.com/libdns/gandi/blob/master/client.go#L20) I'm not so proud of and use either `wiki.gconfs.fr.` and `wiki` to manipulate zones. Can you make this change also ?

obynio commented:
I think I've listed everything that is needed to make the plugin works with the new endpoint.
Feel free to drop me a line if you need help.

obynio commented:
@Themimitoof any news on this ?

Themimitoof commented:
Hello @obynio, unfortunately, I don't have enough bandwidth lately. I hope to be able to work on it this weekend.

Have you a test procedure to try to test locally before requesting you a new review?

obynio commented:
We are lucky enough to have a very easy way to spin out a dev environment using `xcaddy`. It allows you to build and run caddy with your development plugin in two commands. https://github.com/caddyserver/xcaddy#for-plugin-development

You can configure a simple `Caddyfile` like this one. Replace the domain by one you own.
```
pad.leon.re {
	tls {
		dns gandi {env.GANDI_API_TOKEN}
	}
	respond "Hello, world!"
}
```

Then in the directory of the plugin you're working on run `xcaddy`
```
$ xcaddy run --config Caddyfile
```

Then `xcaddy` will fetch the caddy sources, compile it with your plugin and run immediately.
If you need help feel free to reach out to me 👍

Commit:
Improve: Make the code working with the new LiveDNS API

Themimitoof commented on client.go:
```diff
@@ -15,11 +15,10 @@ func (p *Provider) setRecord(ctx context.Context, zone string, record libdns.Rec
 
-	// there is some strange regexp in the api that prevents us from searching names ending with a dot
-	// so we must ensure the names of the record is relative and does not end with a dot
-	recRelativeName := strings.TrimRight(strings.TrimSuffix(record.Name, domain.Fqdn), ".")
```
In fact no we don't have a strange regexp in the api but this is a normal behavior.

When you specify this record in your BIND zone:
```
hello.skusku.pw    IN   A    3600    127.0.0.1
```
The record will be available in `hello.skusku.pw.skusku.pw` because we wasn't added the point to mark the end of the FQDN (also named `absolute record` in the RFC1035 (*Domain names that end in a dot are called
absolute, and are taken as complete*).

I think you already know that, just as reminder ;)

Themimitoof commented on README.md:
```diff
@@ -7,20 +7,20 @@ This package implements the [libdns interfaces](https://github.com/libdns/libdns
 ## Authenticating
 
-This package supports **API Key authentication** but does not yet support **Sharing ID authentication**. Refer to the [LiveDNS documentation](https://doc.livedns.gandi.net/) for more information.
+This package only supports **API Key authentication**. Refer to the [Gandi's Public API documentation](https://api.gandi.net/docs/reference/#Authentication) for more information.
```
Sharing ids are not an authentication mechanism. In fact a sharing id is the identifier of an organization.

Themimitoof commented:
Hello @obynio :wave: 

Here is my modifications to make the module working this the new LiveDNS API. I tested a couple of times and adding/append and deleting records works fine.

It was the first time I played with Go, it was fun to understand how the module works and to contribute to it 😅

> The documentation explicitly states Please note that this API is currently in BETA testing, so care should be taken when used in production

Soon it will disappear, don't worry about that 😉
